src/main/webapp/app/exercises/shared/exercise-headers/header-exercise-page-with-details.component.ts

```diff
@@ -26,24 +25,30 @@ export class HeaderExercisePageWithDetailsComponent implements OnInit, OnChanges
 
     icon: IconProp;
 
-    constructor(private exerciseService: ExerciseService) {}
+    constructor() {}
 
     /**
      * Sets the status badge and categories of the exercise on init
      */
     ngOnInit(): void {
         this.setExerciseStatusBadge();
-        this.exerciseCategories = this.exercise.categories || [];
-        this.setIcon(this.exercise.type);
+        this.exerciseCategories = this.exercise?.categories || [];
+
+        if (this.exercise) {
+            this.setIcon(this.exercise.type);
+        }
```

Comment:
Same with ngOnChanges so can be moved into a function

Reply:
yes please avoid duplicated code and extract a method